package/metrics_daemon/metrics_daemon/src/metrics_daemon.c

```diff
@@ -0,0 +1,378 @@
+/*
+ * Copyright (C) 2018 Swift Navigation Inc.
+ * Contact: Swift Navigation <dev@swiftnav.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#include <getopt.h>
+#include <string.h>
+#include <unistd.h>
+#include <ftw.h>
+#include <json-c/json.h>
+#include <libpiksi/logging.h>
+#include <libpiksi/settings.h>
+#include <libpiksi/util.h>
+
+#define PROGRAM_NAME "metrics_daemon"
+
+#define METRICS_ROOT_DIRECTORY "/var/log/metrics"
+#define METRICS_OUTPUT_FILENAME "/var/log/metrics.json"
+#define METRICS_OUTPUT_FULLPATH METRICS_ROOT_DIRECTORY METRICS_OUTPUT_FILENAME
+#define METRICS_USAGE_UPDATE_INTERVAL_MS (10000u)
+
+#define MAX_FOLDER_NAME_LENGTH 64
+static json_object *jobj_root = NULL;
+static json_object *jobj_cur = NULL;
+static char *root_name = NULL;
+static unsigned int root_length = 0;
+static bool first_folder = true;
+static char *target_file = METRICS_OUTPUT_FILENAME;
+const char *metrics_path = METRICS_ROOT_DIRECTORY;
+int handle_walk_path(const char *fpath, const struct stat *sb, int tflag);
+char *extract_filename(const char *str);
+json_object *init_json_object(const char *path);
+/**
+ * @brief function updates json to file
+ *
+ * @param root       pointer to the root of json object
+ * @param file_name  file to write to
+ */
+static int write_json_to_file(struct json_object *root, const char *file_path)
+{
+  FILE *fp = fopen(file_path, "w+");
+  if (fp == NULL) {
+    piksi_log(LOG_ERR, "error opening %s", file_path);
+    return -1;
+  }
+
+  bool success = (fprintf(fp, "%s\n", json_object_to_json_string(root)) > 0);
+
+  fclose(fp);
+
+  if (!success) {
+    piksi_log(LOG_ERR, "error writing %s", file_path);
+    return -1;
+  }
+
+  return 0;
+}
+
+/**
+ * @brief static function that returns file name from path
+ *
+ * @param  str        file path in ASCII
+ * @return file_name  file name
+ */
+char *extract_filename(const char *str)
```

Comment:
Let's try to attribute code if taken directly from external source if possible. e.g. https://github.com/swift-nav/piksi_buildroot/blob/e47e15952ab3c37e28497b14806f8bc2f38198f3/package/cell_modem_daemon/src/at_command_utils.c#L49-L79
or https://github.com/swift-nav/piksi_buildroot/blob/b2c5d84da2e1a488434dfa4ac3130f515865cbf8/package/libpiksi/libpiksi/src/interface.c#L145-L189